compiler/ilgen/IlBuilder.cpp

```diff
@@ -1419,6 +1419,54 @@ IlBuilder::Call(const char *functionName, int32_t numArgs, TR::IlValue ** argVal
    return NULL;
    }
 
+/**
+ * This service is created for incrementing the value
+ * of address+offset postion so the input "address"
+ * has to be TR::Address type.
+ * atomicAdd32BitSymbol can generate "lock add" in codegen.
+ * This service is specifically for Int32.
+ */
+TR::IlValue *
+IlBuilder::AtomicIntegerAddWithOffset(TR::IlValue * address, TR::IlValue * offset, TR::IlValue * value)
+   {
+   TR_ASSERT(address->getSymbol()->getDataType() == TR::Address, "The IlType of first params must be address");
+   TraceIL("IlBuilder[ %p ]::AtomicIntegerAddWithOffset (%d, %d, %d)\n", this, address->getCPIndex(), offset->getCPIndex(), value->getCPIndex());
+   appendBlock();
+
+   TR::DataType returnType = TR::Int32;
+   
+   TR::SymbolReference *methodSymRef = symRefTab()->findOrCreateCodeGenInlinedHelper(TR::SymbolReferenceTable::atomicAdd32BitSymbol); //lock add
+   TR::Node *callNode;
+   if(offset==NULL){
```

Comment:
I included these two lines because there is a little difference between two call creations (2 vs. 3 for number of nodes) but I changed it to make it more clear. Now I only left setting child in the if()